be/src/Project/Services/DownloadCenter/DownloadCenter.Api/Services/DownloadCenter/DownloadCenterService.cs

```diff
@@ -0,0 +1,350 @@
+            }
+
+            var subcategories = this.context.Categories.Where(x => x.ParentCategoryId == downloadCenterCategory.Id && x.IsVisible && x.IsActive);
+
+            var downloadCenterSubcategories = new List<DownloadCenterSubcategoryServiceModel>();
+
+            foreach (var subcategory in subcategories.OrEmptyIfNull().ToList())
+            {
+                var subcategoryItem = new DownloadCenterSubcategoryServiceModel
+                {
+                    Id = subcategory.Id
+                };
+
+                var subcategoryTranslation = this.context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == subcategory.Id && x.IsActive && x.Language == model.Language);
+
+                if (subcategoryTranslation is null)
+                {
+                    subcategoryTranslation = this.context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == subcategory.Id && x.IsActive);
+                }
+
+                subcategoryItem.Name = subcategoryTranslation?.Name;
+
+                downloadCenterSubcategories.Add(subcategoryItem);
+            }
+
+            var files = this.context.CategoryFiles.Where(x => x.CategoryId == downloadCenterCategory.Id && x.IsActive);
+
+            if (files.Any())
+            {
+                category.Files = files.Select(x => x.MediaId);
+            }
+
+            category.Subcategories = downloadCenterSubcategories;
+
+            return category;
+        }
+
+        public async Task<Guid> UpdateAsync(UpdateDownloadCenterFileServiceModel model)
+        {
+            var downloadCenterFiles = this.context.CategoryFiles.Where(x => x.MediaId == model.Id && x.IsActive);
+
+            if (downloadCenterFiles is null)
+            {
+                throw new CustomException(this.downloadCenterLocalizer.GetString("DownloadCenterFilesNotFound"), (int)HttpStatusCode.NotFound);
+            }
+
+            foreach(var downloadCenterFile in downloadCenterFiles.OrEmptyIfNull())
+            {
+                this.context.CategoryFiles.Remove(downloadCenterFile);
+            }
+
+            foreach (var categoryId in model.CategoriesIds.OrEmptyIfNull())
+            {
+                foreach(var downloadCenterFile in model.Files.OrEmptyIfNull())
+                {
+                    var file = new CategoryFile
+                    {
+                        MediaId = downloadCenterFile.Id,
+                        Name = downloadCenterFile.Name,
+                        CategoryId = categoryId
+                    };
+
+                    await this.context.CategoryFiles.AddAsync(file.FillCommonProperties());
+                }
+            }
+
+            await this.context.SaveChangesAsync();
+
+            return model.Files.FirstOrDefault().Id;
+        }
+
+        public async Task UpdateFileNameAsync(Guid? id, string name)
+        {
+            var file = await this.context.CategoryFiles.FirstOrDefaultAsync();
```

Comment:
**suggestion:** there is for sure the condition in the FirstOrDefaultAsync missing. At the moment it takes always the first file available.

Reply:
You've right